tests/0113-sticky_partitioning.cpp

```diff
@@ -0,0 +1,196 @@
+        read_val(d, tostr() << "/topics/" << topic_ << "/partitions/" 
+                 << i << "/msgs", partition_msgcnt[i]);
+        if(partition_msgcnt[i] > 1) num_partitions_active++;
+    }
+  }
+};
+
+
+/**
+ * @brief Specify sticky.partitioning.linger.ms and parse stats 
+ * to verify it takes effect.
+ */
+static void do_test_sticky_partitioning (bool sticky_delay_long) {
+  
+  std::string topic = Test::mk_topic_name(__FILE__, 0);
+  Test::create_topic(NULL, topic.c_str(), 3, 1);
+  myParStats event_cb(topic);
+
+  RdKafka::Conf *conf;
+  Test::conf_init(&conf, NULL, 0);
+
+  const int msgcnt = 50000;
+  const int msgsize = 1000;
+
+  if (sticky_delay_long){
+        Test::conf_set(conf, "sticky.partitioning.linger.ms", "100000");
+  }else{
+        Test::conf_set(conf, "sticky.partitioning.linger.ms", "1");
+  }
+
+  Test::conf_set(conf, "statistics.interval.ms", "4000");
+  std::string errstr;
+  if (conf->set("event_cb", &event_cb, errstr) != RdKafka::Conf::CONF_OK)
+    Test::Fail(errstr);
+
+  RdKafka::Producer *p = RdKafka::Producer::create(conf, errstr);
+  if (!p)
+          Test::Fail("Failed to create Producer: " + errstr);
+
+  /* Produce messages */
+  char val[msgsize];
+  memset(val, 'a', msgsize);
+
+  for (int i = 0 ; i < msgcnt ; i++) {
+    RdKafka::ErrorCode err = p->produce(topic, RdKafka::Topic::PARTITION_UA,
+                                        RdKafka::Producer::RK_MSG_COPY,
+                                        val, msgsize, NULL, 0, -1, NULL);
+    if (err)
+      Test::Fail("Produce failed: " + RdKafka::err2str(err));
+  }
+
+  Test::Say(tostr() << "Produced " << msgcnt << " messages\n");
+  p->flush(5*1000);
+
+  Test::Say("Waiting for stats\n");
+  while (event_cb.num_partitions_active == 0)
+    p->poll(1000);
+
+  Test::Say("Partition Message Count: \n");
+  for(int i = 0; i < 3; i++){
+        Test::Say(tostr() << " " << i << ": " << 
+                  event_cb.partition_msgcnt[i] << "\n");
+  }
+
+  /* When sticky.partitioning.linger.ms is long (greater than 
+   * length of run), one partition should receive messages. */
+  if (sticky_delay_long  &&
+      event_cb.num_partitions_active > 1)
+    Test::Fail(tostr() 
+               << "Expected only 1 partition to receive msgs" 
+               << " but " << event_cb.num_partitions_active 
+               << " partitions received msgs.");
+  
+  /* When sticky.partitioning.linger.ms is short (sufficiently
+   * smaller than length of run), extremely likely that all  
+   * partitions receive messages. */
+  if (!sticky_delay_long  &&
+      event_cb.num_partitions_active < 3)
+    Test::Fail(tostr() 
+               << "Expected all partitions to receive msgs" 
+               << " but only " << event_cb.num_partitions_active
+               << " partitions received msgs.");
+
+  delete p;
+
+}
+#endif
+
+extern "C" {
+  int main_0113_sticky_partitioning (int argc, char **argv) {
+#if WITH_RAPIDJSON
+    do_test_sticky_partitioning(true);
```

Comment:
nit: best to avoid bool args as they're not self descriptive, or you can put an inline comment `/*long delay*/`, `/*short delay*/` also fine.